Corrfunc/tests/common.py

```diff
@@ -0,0 +1,72 @@
+import os
+from os.path import dirname, abspath, join as pjoin
+
+import numpy as np
+import pytest
+
+from Corrfunc.io import read_fastfood_catalog
+
+@pytest.fixture(scope='module')
+def gals_Mr19():
+    
+    filename = pjoin(dirname(abspath(__file__)),
+                    "../../theory/tests/data", "gals_Mr19.ff")
+    x, y, z, w = read_fastfood_catalog(filename, need_weights=True)
+    
+    return x, y, z, w
+
+
+@pytest.fixture(scope='module')
+def Mr19_mock_northonly():
+    filename = pjoin(dirname(abspath(__file__)),
+                "../../mocks/tests/data", "Mr19_mock_northonly.rdcz.ff")
+    ra,dec,cz,w = read_fastfood_catalog(filename, need_weights=True)
+    
+    return ra, dec, cz, w
+
+
+@pytest.fixture(scope='module')
+def Mr19_randoms_northonly():
+    filename = pjoin(dirname(abspath(__file__)),
+                "../../mocks/tests/data", "Mr19_randoms_northonly.rdcz.ff")
+    ra,dec,cz,w = read_fastfood_catalog(filename, need_weights=True)
+    
+    return ra, dec, cz, w
+
+
+def maxthreads():
+    '''Use as many threads as cores that are available to this process'''
+    import multiprocessing
+    
+    try:
+        maxthreads = len(os.sched_getaffinity(0))
+    except:
+        maxthreads = multiprocessing.cpu_count() or 1
+        
+    return maxthreads
+
+
+def all_isa_nthreads(fastest_nthreads=4):
+    '''Test all ISA with maxthreads, and the fastest ISA with threads 1 to `fastest_nthreads`'''
+    mx = maxthreads()
+    # don't test with more threads than cores
+    fastest_nthreads = min(fastest_nthreads,mx)
+    # ... except for one test, which we'll force to have more threads than cores as a "stress test"
+    all_nthreads = list(range(1,fastest_nthreads+1)) + [mx+1,]
```

Comment:
This is PEP compatible, right? 

I would also suggest renaming fastest_nthreads to max_nthreads - which (I think) is what you are after. Additionally, line 55 is doing too much by itself - I am not sure I understand what's happening. Perhaps, split this into two lines to show that the normal usage about testing for all threads `[1, fastest_nthreads]` and then append to this list to show an additional `nthreads` test is done.

This function has a high cognitive load. From what I see this function is generating all allowed combinations of `ISAs` and `nthreads` (a list of `[1, max_nthreads]`) - therefore, the function name might be better off as `generate_isa_and_nthreads_combos` (in keeping with the convention of using a verb for a function name).

Reply:
> I would also suggest renaming fastest_nthreads to max_nthreads - which (I think) is what you are after
`fastest_nthreads` is the number of threads that we will use in the thread sweep, which only uses the fastest ISA. This will likely not be the same as the number of threads used in the ISA sweep! Agreed this is a subtle point, let me refactor this...